packages/babel-traverse/src/scope/index.js

```diff
@@ -740,6 +743,10 @@ export default class Scope {
     path.traverse(collectorVisitor, state);
     this.crawling = false;
 
+    if (state.unnamedDefault) {
+      state.unnamedDefault.id = this.generateUidIdentifier("default");
```

Comment:
https://github.com/babel/babel/pull/5956 seems like another case that is a good example where people will make assumptions, but should instead update their code to match, rather than having `babel-traverse` auto-inject the catch binding for consistency. People will absolutely assume all catch blocks have bindings, and they'll have to learn not to assume that if they want their transform to work on all inputs.